cvat/apps/annotation/annotation.py

```diff
@@ -77,13 +78,59 @@ def serialize(self):
         if serializer.is_valid(raise_exception=True):
             return serializer.data
 
+    @staticmethod
+    def _is_shape_inside(shape, start, stop):
+        return start <= int(shape['frame']) <= stop
+
+    @staticmethod
+    def _is_track_inside(track, start, stop):
+        # a <= b
+        def has_overlap(a, b):
+            return 0 <= min(b, stop) - max(a, start)
+
+        prev_shape = None
+        for shape in track['shapes']:
+            if prev_shape and not prev_shape['outside'] and \
+                has_overlap(prev_shape['frame'], shape['frame']):
+                    return True
+            prev_shape = shape
+
+        if not prev_shape['outside'] and prev_shape['frame'] <= stop:
+            return True
+
+        return False
+
+    @staticmethod
+    def _slice_track(track_, start, stop):
+        track = copy.deepcopy(track_)
+        segment_shapes = [s for s in track['shapes'] if AnnotationIR._is_shape_inside(s, start, stop)]
+
+        if len(segment_shapes) < len(track['shapes']):
+            interpolated_shapes = TrackManager.get_interpolated_shapes(track, start, stop)
+
+            for shape in interpolated_shapes:
+                if shape['frame'] == start and \
+                    (not segment_shapes or segment_shapes[0]['frame'] > start):
+                    segment_shapes.insert(0, shape)
+                elif shape['frame'] == stop and \
+                    (not segment_shapes or segment_shapes[-1]['frame'] < stop):
+                    segment_shapes.append(shape)
+            drop_shape_count = sum(1 for _ in itertools.takewhile(lambda s: s['outside'], segment_shapes))
+            segment_shapes = segment_shapes[drop_shape_count:]
+            del track['interpolated_shapes']
+            for shape in segment_shapes:
+                del shape['keyframe']
```

Comment:
Why should we delete the property for all segment_shapes? Could you please add a comment?